File: Data Structures & Algorithms/string-encode-and-decode/submission-0.cs
public class Solution {

    public string Encode(IList<string> strs) {
        var res = new StringBuilder();
        foreach (var str in strs)
        {
            res.Append(str.Length + "#" + str);
        }
        return res.ToString();        
    }

    public List<string> Decode(string s) {
        var res = new List<string>();

        string occur = "";
        var ind = 0;
        while (ind < s.Length)
        {
            if (Char.IsDigit(s[ind]))
            {
                occur += s[ind].ToString();
                ind++;
            }
            else if (s[ind] - '0' == '#' - '0')
            {
                var occurNum = Int32.Parse(occur);
                occur = "";
                res.Add(s.Substring(ind + 1,occurNum));
                ind = ind + 1 + occurNum;
            }
        }
        return res;
   }
}

File: Data Structures & Algorithms/basic-calculator-ii/submission-0.py
class Solution:
    def calculate(self, s: str) -> int:
        cur = prev = res = 0

        i = 0
        curr_char = '+'

        while i < len(s):

            if (s[i].isdigit()):
                while i < len(s) and s[i].isdigit():
                    cur = cur * 10 + int(s[i])
                    i += 1
                i -= 1

                if curr_char == '+':
                    res += cur
                    prev = cur
                elif curr_char == '-':
                    res -= cur
                    prev = -cur
                elif curr_char == '*':
                    res -= prev

                    res += prev * cur
                    prev = prev * cur
                else:
                    res -= prev
                    res += int (prev/cur)
                    prev = int (prev/cur)

                cur = 0
            elif s[i] != ' ':
                curr_char = s[i]

            i += 1

        return res
        